src/apps/api/views/analytics.py

```diff
@@ -0,0 +1,173 @@
+                name='start_date',
+                location='query',
+                required=True,
+                type='string',
+                description='Beginning of query interval (inclusive) (YYYY-MM-DD format string)'
+            ),
+            coreapi.Field(
+                name='end_date',
+                location='query',
+                required=True,
+                type='string',
+                description='End of query interval (exclusive) (YYYY-MM-DD format string)'
+            ),
+            coreapi.Field(
+                name='time_unit',
+                location='query',
+                required=True,
+                type='string',
+                description='Unit of time (choose 1 of month, week, or day)'
+            ),
+            coreapi.Field(
+                name='format',
+                location='query',
+                required=False,
+                type='string',
+                description='If csv data is desired set format=csv, otherwise do not set.'
+            ),
+        ]
+
+        return fields
+
+def merge_dicts(d1, d2):
+    d = {**d1, **d2}
+    return d
+
+def build_request_object(model_name, filter_param, time_unit, start_date, end_date, csv, data_key, count_key):
+    filter_args = {
+        filter_param + '__range': [start_date, end_date]
+    }
+
+    if csv:
+        output_fields = {
+            count_key: Count('pk'),
+            data_key: F('datefield')
+        }
+        data = model_name.objects.filter(**filter_args).dates(filter_param, time_unit).values(**output_fields)
+    else:
+        data = model_name.objects.filter(**filter_args).dates(filter_param, time_unit).values(count=Count('pk'), _datefield=F('datefield'))
+    return data
+
+
+class AnalyticsRenderer(r.CSVRenderer):
+    header = [
+            'start_date',
+            'end_date',
+            'time_unit',
+            'registered_user_count',
+            'competition_count',
+            'competitions_published_count',
+            'submissions_made_count',
+            'users_data_date',
+            'users_data_count',
+            'competitions_data_date',
+            'competitions_data_count',
+            'submissions_data_date',
+            'submissions_data_count',
+        ]
+
+    labels = {
+        'start_date': 'Start Date',
+        'end_date': 'End Date',
+        'time_unit': 'Time Unit',
+        'registered_user_count': 'Registered User Count',
+        'competition_count': 'Competition Count',
+        'competitions_published_count': 'Competitions Published Count',
+        'submissions_made_count': 'Submissions Made Count',
+        'users_data_date': 'Users Data Date',
+        'users_data_count': 'Users Data Count',
+        'competitions_data_date': 'Competitions Data Date',
+        'competitions_data_count': 'Competitions Data Count',
+        'submissions_data_date': 'Submissions Data Date',
+        'submissions_data_count': 'Submissions Data Count',
+    }
```

Comment:
can get the headers w/ `header = labels.keys()` (may have to force cast to a list?)

Reply:
Done.